cmd/kube-apiserver/app/server.go

```diff
@@ -319,6 +320,16 @@ func CreateKubeAPIServerConfig(
 		}
 	}
 
+	if genericConfig.EgressSelector != nil {
```

Comment:
Not sure if we should replace the dialer here since line 282 (https://github.com/kubernetes/kubernetes/pull/84381/files#diff-36e1d7088209b1ab6ff878b18bb35210R282) builds the genericConfig with the ssh tunnel dialer while we replace it in extraconfig.

I'd recommend replacing the dialer earlier to avoid ambiguity.

cc: @caesarxuchao

Reply:
ah yeah I remeber why I set it there :sweat_smile: should it just return where it was then?

Reply:
hmm but it seems it doesn't change anything :thinking: not sure which is best

Reply:
In `buildGenericConfig`, the proxyTransport is used to build the serviceResolver used by the (dynamic) audit webhooks and the admission webhooks. Those cases are covered by #85168.

Actually moving this code block to line [379](https://github.com/kubernetes/kubernetes/blob/789dc873f6816cc2b9b39e77a9b94f478d3a3134/cmd/kube-apiserver/app/server.go#L379) is even better.

Reply:
done :)